NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Common/XsqldaMarshaler.cs

```diff
@@ -21,58 +21,33 @@
 
 using System;
 using System.Runtime.InteropServices;
-using System.Text;
-
 using FirebirdSql.Data.Common;
+using System.IO;
 
 namespace FirebirdSql.Data.Client.Common
 {
-	internal sealed class XsqldaMarshaler
+	internal static class XsqldaMarshaler
 	{
-		#region Static Fields
-
-		private static readonly XsqldaMarshaler instance = new XsqldaMarshaler();
-
-		#endregion
-
-		#region Static Properties
+		#region � Methods �
 
-		public static XsqldaMarshaler Instance
-		{
-			get { return XsqldaMarshaler.instance; }
-		}
-
-		#endregion
-
-		#region Constructors
-
-		private XsqldaMarshaler()
-		{
-		}
-
-		#endregion
-
-		#region Methods
-
-		public void CleanUpNativeData(ref IntPtr pNativeData)
+		public static void CleanUpNativeData(ref IntPtr pNativeData)
 		{
 			if (pNativeData != IntPtr.Zero)
 			{
 				// Obtain XSQLDA information
-				XSQLDA xsqlda = new XSQLDA();
 
-				xsqlda = (XSQLDA)Marshal.PtrToStructure(pNativeData, typeof(XSQLDA));
+				var xsqlda = (XSQLDA)Marshal.PtrToStructure(pNativeData, typeof(XSQLDA));
 
 				// Destroy XSQLDA structure
 				Marshal.DestroyStructure(pNativeData, typeof(XSQLDA));
 
 				// Destroy XSQLVAR structures
-				for (int i = 0; i < xsqlda.sqln; i++)
+				for (var i = 0; i < xsqlda.sqln; i++)
 				{
-					IntPtr ptr1 = GetIntPtr(pNativeData, ComputeLength(i));
+					var ptr1 = GetIntPtr(pNativeData, ComputeLength(i));
 
 					// Free	sqldata	and	sqlind pointers	if needed
-					XSQLVAR sqlvar = (XSQLVAR)Marshal.PtrToStructure(ptr1, typeof(XSQLVAR));
+					var sqlvar = (XSQLVARCleanup)Marshal.PtrToStructure(ptr1, typeof(XSQLVARCleanup));
```

Comment:
What about creating `XSQLVAREmpty` (or something like that) and from that deriving `XSQLVAR`? Then all the `sqltype` etc. fields will not be repeated and it can be just one place.